Document/CategoryManager.php

```diff
@@ -11,14 +11,127 @@
 
 namespace Sonata\ClassificationBundle\Document;
 
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Sonata\ClassificationBundle\Model\CategoryInterface;
 use Sonata\ClassificationBundle\Model\CategoryManagerInterface;
+use Sonata\ClassificationBundle\Model\ContextInterface;
+use Sonata\ClassificationBundle\Model\ContextManagerInterface;
 use Sonata\CoreBundle\Model\BaseDocumentManager;
 use Sonata\DatagridBundle\Pager\Doctrine\Pager;
 use Sonata\DatagridBundle\ProxyQuery\Doctrine\ProxyQuery;
 
 class CategoryManager extends BaseDocumentManager implements CategoryManagerInterface
 {
     /**
+     * @var CategoryInterface[]
+     */
+    protected $categories;
+
+    /**
+     * @var ContextManagerInterface
+     */
+    protected $contextManager;
+
+    /**
+     * @param string                  $class
+     * @param ManagerRegistry         $registry
+     * @param ContextManagerInterface $contextManager
+     */
+    public function __construct($class, ManagerRegistry $registry, ContextManagerInterface $contextManager)
+    {
+        parent::__construct($class, $registry);
+
+        $this->contextManager = $contextManager;
+        $this->categories = array();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRootCategoriesPager($page = 1, $limit = 25, $criteria = array())
+    {
+        $page = (int) $page == 0 ? 1 : (int) $page;
```

Comment:
Remove the line, because we never check the `$page` parameters in any method. This is the task of the `Pager`.